keras_cv/losses/giou_loss.py

```diff
@@ -0,0 +1,154 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import tensorflow as tf
+
+from keras_cv import bounding_box
+
+
+class GIoULoss(tf.keras.losses.Loss):
+    """Implements the GIoU Loss
+    GIoU loss is a modified IoU loss commonly used for object detection. This loss aims
+    to directly optimize the IoU score between true boxes and predicted boxes. GIoU loss
+    adds a penalty term to the IoU loss that takes in account the area of the
+    smallest box enclosing both the boxes being considered for the iou. The length of
+    the last dimension should be 4 to represent the bounding boxes.
+
+    Args:
+        bounding_box_format: a case-insensitive string (for example, "xyxy").
+            Each bounding box is defined by these 4 values.For detailed information
+            on the supported formats, see the
+            [KerasCV bounding box documentation](https://keras.io/api/keras_cv/bounding_box/formats/).
+        axis: the axis along which to mean the ious. Passing the string "no_reduction" implies
+            mean across no axes. Defaults to -1.
```

Comment:
I prefer None = do not reduce on any axis here. Otherwise we're not consistent with how reduction works with TF/Keras losses

Reply:
`None` implies reduction across all axes in Keras losses. That's why I added a new string specifically for the use case of YoloX.

Reply:
axis=None actually means axis=-1:
https://github.com/tensorflow/tensorflow/blob/d5b57ca93e506df258271ea00fc29cf98383a374/tensorflow/python/ops/nn_ops.py#L4106-L4107

The point here is that taking a string is quite odd, can we remove `axis` from this PR, and add a TODO to unify axis naming convention together with `keras.losses.Huber` later? (I believe that loss also needs to take some reduction axis as well)

Reply:
Sure, I'll remove this for now and add an issue.